### PR TITLE
Add Quickstart Examples

### PR DESCRIPTION
**Pull request by vfordpivotal**
* Adds 3 examples with additional documentation
* Basic put/get example
* User defined classes and PDX
* Using new GF 10+ JSON document

### CODE REVIEW

**Review by charliemblack — APPROVED** [review]
Ship it

File: quickstart/001_client_server_introduction/README.md [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+
+VMware GemFire is an in-memory distributed Key-Value datastore.  As a datastore, GemFire provides a real-time, consistent and distributed service for modern applications with data-intensive needs and low latency response requirements. Because of GemFire's distributed peer-to-peer nature it can take advantage of multiples servers to pool memory, cpu and disk storage for improved performance, scalability and fault tolerance to build applications needing caching, management of in-flight data or the key-value database of record.
+
+## Goal
+
+The goal of this quickstart tutorial is to introduce GemFire basics, including starting a GemFire development environment, building and connecting a Java based GemFire client application to a cluster and performing basic CRUD operations.
+
+## Download Examples and Configure Environment
+
+Download and install VMware GemFire from [Tanzu Network](https://network.pivotal.io). Follow the installation instructions in the [GemFire documentation](https://docs.vmware.com/en/VMware-Tanzu-GemFire/index.html).
```
Comment (davebarnes97) [inline quickstart/001_client_server_introduction/README.md:11]: "VMware-Tanzu-GemFire/index.html" -> "VMware-GemFire/index.html"

File: quickstart/001_client_server_introduction/README.md [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+
+Clone the GemFire examples repository from GitHub.
+
+```text
+$ git clone git@github.com:gemfire/gemfire-examples.git
+```
+
+Set Gemfire home environment variable to top of GemFire install directory. Note for this example Gemfire is installed in the home directory of the user - adjust as necessary for local environment and install directory location.
```
Comment (davebarnes97) [inline quickstart/001_client_server_introduction/README.md:19]: "Set Gemfire home environment variable to top..." -> "Set the GEMFIRE_HOME environment variable to the top..."

File: quickstart/001_client_server_introduction/README.md [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+------- | -------------------------------------------------------------------
+locator | test-javaclient(locator:532579:locator)<ec><v0>:41000 [Coordinator]
+server | test-javaclient(server:534075)<v1>:41001
+
+```
+
+The above response of the "list members" command should show two members in the cluster.
+
+There are multiple useful commands provided by gfsh to configure and manage GemFire clusters,
+the *help* command will list them and the gfsh interactive shell has tab completion to help with determining options for each command. For additional information checkout the [GFSH documation](https://docs.vmware.com/en/VMware-Tanzu-GemFire/9.15/tgf/GUID-tools_modules-gfsh-chapter_overview.html) .
```
Comment (davebarnes97) [inline quickstart/001_client_server_introduction/README.md:185]: "https://docs.vmware.com/en/VMware-Tanzu-GemFire/9.15/tgf/GUID-tools_modules-gfsh-chapter_overview.html" -> "https://docs.vmware.com/en/VMware-GemFire/9.15/gf/tools_modules-gfsh-chapter_overview.html"

File: quickstart/001_client_server_introduction/README.md [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+
+## Create a server Region
+
+A [Region](https://docs.vmware.com/en/VMware-Tanzu-GemFire/9.15/tgf/GUID-developing-region_options-chapter_overview.html) is the core structure in the server for holding and managing key-value data in GemFire, it is similar to a hashmap but is distributed across the server members.
```
Comment (davebarnes97) [inline quickstart/001_client_server_introduction/README.md:189]: "https://docs.vmware.com/en/VMware-Tanzu-GemFire/9.15/tgf/GUID-developing-region_options-chapter_overview.html" -> "https://docs.vmware.com/en/VMware-GemFire/9.15/gf/developing-region_options-chapter_overview.html"

File: quickstart/001_client_server_introduction/README.md [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+
+While still in the gfsh interactive shell, create a [partition region](https://docs.vmware.com/en/VMware-Tanzu-GemFire/9.15/tgf/GUID-developing-partitioned_regions-chapter_overview.html).
```
Comment (davebarnes97) [inline quickstart/001_client_server_introduction/README.md:191]: "https://docs.vmware.com/en/VMware-Tanzu-GemFire/9.15/tgf/GUID-developing-partitioned_regions-chapter_overview.html" -> "https://docs.vmware.com/en/VMware-GemFire/9.15/gf/developing-partitioned_regions-chapter_overview.html"

**Review by davebarnes97 — CHANGES_REQUESTED** [review]
@vfordpivotal Nice work - docs and examples, both.
I plan to run these myself and provide feedback, but due to schedules I may not be able to start for a week or more. So if you'd like, you can merge these without waiting for me. I'll run them later and submit tickets if I find issues.
Before you merge, please update the doc URLs. They changed out from under you while you were developing these examples. The new pattern has 3 differences from the old one:

- "VMware-Tanzu-GemFire" is now "VMware-GemFire"
- "tgf" is now "gf"
- the "GUID-" string (a DocWorks artifact) is gone
So, for example:
"https://docs.vmware.com/en/VMware-Tanzu-GemFire/9.15/tgf/GUID-developing-region_options-chapter_overview.html" -> "https://docs.vmware.com/en/VMware-GemFire/9.15/gf/developing-region_options-chapter_overview.html"
There are a few more subtleties when it comes to the GF-for-TAS (GF-for-VMs) doc -- LMK if you'd like the complete mapping for the entire doc set.

**Review by metatype — APPROVED** [review]

**Pull request merged**